content/core-concepts/tracing.md

```diff
@@ -4,48 +4,49 @@ weight: 20
 ---
 
 
-A trace tracks the progression of a single user request
+Tracing tracks the progression of a single user request
 as it is handled by other services that make up an application.
 
 Each unit work is called a span in a trace. Spans include metadata about the work,
 including the time spent in the step (latency) and status.
 You can use tracing to debug errors and
 latency issues of your application.
 
-#### Spans
+#### Span
 
 A trace is a tree of spans.
 
 A span is the unit work represented in a trace. A span may
 represent a HTTP request, an RPC, a server handler,
 a database query or a section customly marked in user code.
 
+For example:
+
 ![A trace](/img/trace-trace.png)
 
 Above, you see a trace with various spans. In order to respond
-to `/messages`, several other internal requests are made. First,
-we are checking if the user is authenticated, we are trying to
-get the results from the cache. It is a cache miss, hence we
-query the database for the results, we cache the results back,
-and respond back to the user.
-
-There are two types of spans:
+to `/messages`, several other internal requests are made. Firstly,
+we check if the user is authenticated. Next we check if their
+messages were cached. Since their message wasn't cached, that's
+a cache miss and we then fetch their content from MySQL, cache it
+and then provide the response containing their messages.
 
-* **Root span**: Root spans don't have a parent span. They are the
-  first span. `/messages` span above is a root span.
-* **Child span**: Child spans have an existing span as their parent.
+A span may or may not have a parent span:
 
+* A span without a parent is called a **"root span"** for example span "/messages"
+* A span with a parent is called a **"child span"** for example spans "auth", "cache.Get", "mysql.Query", "cache.Put"
 
-Spans are identified with an ID and are associated to a trace.
-These identifiers and options byte together are called span context.
-Inside the same process, span context is propagated in a context
+Spans are identified with a SpanID and each span belongs to a single trace.
+These identifiers and options byte together are called **Span Context**.
+Inside the same process, **Span context** is propagated in a context
 object. When crossing process boundaries, it is serialized into
-protocol headers. The receiving end can read the span context
-and create child spans.
+protocol headers. The receiving end can read the **Span context** and create child spans.
+
+Each trace is uniquely identified by a TraceID which all constituent spans will share.
```

Comment:
Maybe move this paragraph up (after "each span belongs to a single trace")? Since span context also includes trace id.